nav2_core/include/nav2_core/recovery.hpp

```diff
@@ -30,6 +32,8 @@ namespace nav2_core
 class Recovery
 {
 public:
+  using Ptr = std::shared_ptr<Recovery>;
```

Comment:
`SharedPtr` instead?

Reply:
So this is `Ptr` to be consistent across the other adaptors. Local/global/goal checker use `Ptr`

Reply:
So I was about to just change them all, but is also appears all the DWB plugins and BT  also use ::Ptr to describe shared ptrs, the only place defined in the stack that we do SharedPtr is in the lifecycle node wrapper (and subsequent rclcpp calls) which is a little too much to casually add to this PR. Can we make this out of scope?

Reply:
Yes, we can leave that for a later PR.